README.md

```diff
@@ -249,6 +267,33 @@ npm run prepublishOnly
 
+
+# Web Chat API Reference
+
+There are several properties that you might pass into your Web Chat React Component (`<ReactWebChat>`) or the `renderWebChat()` method. Feel free to examine the source code starting with [`packages/component/src/Composer.js`](https://github.com/Microsoft/BotFramework-WebChat/blob/master/packages/component/src/Composer.js#L378). Below is a short description of the available props.
+
+| Property                   | Description                                                                                                                                                                                                                                                                                                                               |
+|----------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `activityMiddleware`       | A chain of middleware, modeled after [Redux middleware](https://medium.com/@jacobp100/you-arent-using-redux-middleware-enough-94ffe991e6), that allows the developer to add new DOM components on the currently existing DOM of Activities. The middleware signature is the following: `options => next => card => children => next(card)(children)`. |
+| `activityRenderer`         | The "flattened" version of `activityMiddleware`, similar to the [store enhancer](https://github.com/reduxjs/redux/blob/master/docs/Glossary.md#store-enhancer) concept in Redux.                                                                                                                                                |
+| `adaptiveCardHostConfig`   | Pass in a custom Adaptive Cards host config.                                                                                                                                                                                                                                                                                              |
+| `attachmentMiddleware`     | A chain of middleware that allows the developer to add new DOM on attachments. The signature is the following: `options => next => card => next(card)`.                                                                                                                                                                                 |
```

Comment:
Should we specify custom HTML Elements? e.g. "A chain of middleware that allows the developer to add their own custom HTML Elements on attachments."